src/ZoneServer/Network/PacketHandler.cs

```diff
@@ -1292,7 +1292,9 @@ public void CZ_SKILL_GROUND(IZoneConnection conn, Packet packet)
 					return;
 				}
 
-				character.TurnTowards(direction);
+				if (skill.Id != SkillId.Ranger_Strafe)
```

Comment:
So basically, Strafe causes you to jump in the direction you're facing, then take a shot at the selected enemy.  As such, it has to call the skill handler before triggering the turntowards, or your facing direction will be changed to the the target you're trying to shoot and you'll always jump forwards instead.

I agree that this kinda sucks, but I don't see any other way to do it.

Reply:
I guess that's exactly what we were worried about when this line was added, that we may run into a situation where we don't want to forcefully turn the character in the direction sent by the client. There are other ways to solve this, like adding an interface that implements a property or method that lets us customize this behavior for a skill handler that implements it, but frankly, it was probably a mistake to cut corners here in the first place.

It may be time to reconsider our skill handler parameters, and perhaps create an argument class/struct that can encapsulate the information sent by the client and give full control back to the handlers. That way we won't need to worry about arguments changing again. After all, there are still four unknowns in the `CZ_SKILL_GROUND` handler for example, and we'd want neither handle methods with 10 parameters, nor would be want to constantly change our handler definitions when we find use for a previously unused parameter.